### PR TITLE
Add Nightwatch Tests to App

### LINKED ISSUE #42: Add 3 Nightwatch e2e tests, a few Karma tests
Assignment by Gaurav Parajuli:
"For the web page in there please write few unit and at least three JScript/ Java-based selenium test in Nightwatch.".

First time installing and using:
* Material Design Lite: https://getmdl.io/
* WebPack: a module builder: http://webpack.github.io/
* Vue-loader: a loader for WebPack: http://vue-loader.vuejs.org/en/ 
* Nightwatch: http://nightwatchjs.org/ 

... but here it goes! I've only used Selenium WebDriver with Java for the past two years at Fitbit. Some API testing. See
* http://www.tjmaher.com/2015/12/automate-amazon-development-environment.html
* https://www.tjmaher.com/2016/02/restful-testing-with-stripe-brief.html

Last weekend, it was a coding test using Protractor and AngularJS, for the very first time. 
* http://www.tjmaher.com/2016/09/playing-with-protractor-testing.html

### PR DESCRIPTION
**Pull request by tjmaher**
Fixes #42
* Add more tests to already existing spec\test.js and ui-locators.js.
* Fix typo: randPlanetBtn was listed as 'randplanetBtn'

### CODE REVIEW

File: test/e2e/ui-properties/randplanetBtn [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Created by ruxit on 5/19/16.
+ */
+
+module.exports = {
```
Comment (alexkramer) [inline test/e2e/ui-properties/randplanetBtn]: Is this a duplicate of the ui-locators.js file? Either way might be better to convert this to a page object to represent the home page (for our apps we always create page objects). You can see the documentation here:  
http://nightwatchjs.org/guide#page-objects

You can then start your tests above as:

'test something': browser => {
  browser.page.home().navigate()
}

File: test/e2e/specs/test.js [hunk 2]
```diff
@@ -8,12 +8,77 @@ module.exports = {
...
     .url(browser.launchUrl)
-      .waitForElementVisible('#app', 5000)
-      .assert.elementPresent('.randPersonBtn')
+      .waitForElementVisible(ui.dev.app, 5000)
+      .assert.elementPresent(ui.dev.randomBtn.person)
+      .pause(5000)
+      .end()
+  },
+
+  'verify person record has seven table records': function (browser) {
```
Comment (alexkramer) [inline test/e2e/specs/test.js]: Prefer es6 arrow function notation:
'test name': browser => {  }

File: test/e2e/specs/test.js [hunk 3]
```diff
@@ -8,12 +8,77 @@ module.exports = {
...
+  },
+
+  'verify planet button has expected value: public RANDOM PLANET': function (browser) {
+    browser
+      .url(browser.launchUrl)
+      .waitForElementVisible(ui.dev.app, 5000)
+      .setValue(ui.dev.select,ui.dev.selectValue.planet)
+      .click(ui.dev.select)
+      .pause(1000)
+      .keys(['\uE006']) //hits the enter key.
```
Comment (alexkramer) [inline test/e2e/specs/test.js]: Is it possible to use browser.Keys.ENTER instead of this character reference?

File: test/e2e/specs/test.js [hunk 4]
```diff
@@ -8,12 +8,77 @@ module.exports = {
...
+        this.assert.notEqual(result,'')
+      })
+      .end()
+  },
+
+  'verify person button has expected value: face RANDOM PERSON': function (browser) {
+    browser
+      .url(browser.launchUrl)
+      .waitForElementVisible(ui.dev.app, 5000)
+      .getText(ui.dev.randomBtn.person, function(result) {
```
Comment (alexkramer) [inline test/e2e/specs/test.js]: Might be cleaner to use the assert.containsText() method.

File: test/e2e/specs/test.js [hunk 5]
```diff
@@ -8,12 +8,77 @@ module.exports = {
...
+      .pause(5000)
+      .end()
+  },
+
+  'verify person record has seven table records': function (browser) {
+    browser
+      .url(browser.launchUrl)
+      .waitForElementVisible(ui.dev.app, 5000)
+      .assert.elementPresent(ui.dev.randomBtn.person)
+      .assert.elementCount(ui.dev.tableRecord, 7)
```
Comment (alexkramer) [inline test/e2e/specs/test.js]: Would be a better test to capture what the 7 input elements are.

File: test/e2e/ui-properties/randplanetBtn (hunk 1 above)
Comment (tjmaher) [inline test/e2e/ui-properties/randplanetBtn]: I saw that there was the ui-locators.js file, and wasn't sure if I should create my own and possibly duplicate code, or build on what was already in the framework. I'll convert that to a page object.

File: test/e2e/specs/test.js (hunk 3 above)
Comment (tjmaher) [inline test/e2e/specs/test.js]: Ah! Thank you! I was used to doing sendKeys(KEYS.Enter) in Java, but couldn't find a similar method.

File: test/e2e/ui-properties/randplanetBtn (hunk 1 above)
Comment (tjmaher) [inline test/e2e/ui-properties/randplanetBtn]: Hrm. Name, rotation period, orbital period all have the same css selector -- .mdl-textfield__input. If this was Java I could get them all and pick and choose what I wanted by index. (Name would be get(0). Rotation period, get(1), etc.) ... let me see if there is something similar in JavaScript.

File: test/e2e/specs/test.js (hunk 5 above)
Comment (tjmaher) [inline test/e2e/specs/test.js]: Will do!

File: test/e2e/specs/test.js (hunk 4 above)
Comment (tjmaher) [inline test/e2e/specs/test.js]: Ah! That's how that is composed in Nightwatch. I'll do that. Thank you!

File: test/e2e/specs/test.js (hunk 5 above)
Comment (tjmaher) [inline test/e2e/specs/test.js]: Done. Mostly. Groundwork for page objects is there. They aren't hooked up in the tests exactly as they should be. I embedded a css tag in every Vue element, I modeled after what we use at Fitbit: It is "[data-qa="nameTextbox"]' for a textbox on the Person page.

File: test/e2e/specs/test.js (hunk 2 above)
Comment (tjmaher) [inline test/e2e/specs/test.js]: Finally realized why this was not working in the evaluation version of WebStorm. I needed to upgrade the language of JavaScript.

**Pull request merged**